src/js_command_match/keys/add.rs

```diff
@@ -0,0 +1,169 @@
+        command.push(self.public_key.to_owned());
+        command.push("network-config".to_string());
+        command.push(network_id);
+
+        if self.sign_with_ledger {
+            command.push("sign-with-ledger".to_string());
+            command.push("--seed-phrase-hd-path".to_string());
+            command.push(self.ledger_path.to_owned().unwrap_or_default());
+        } else {
+            command.push("sign-with-keychain".to_string());
+        }
+        command.push("send".to_string());
+
+        command
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use clap::Parser;
+
+    #[test]
+    fn add_function_call_access_key_for_all_methods_testnet() {
+        let account_id = "bob.testnet";
+        let access_key = "ed25519:DReZmNmnGhpsYcCFFeYgPsJ9YCm9xH16GGujCPe3KQEq";
+        let contract_id = "example.testnet";
+
+        for i in 0..CONTRACT_ID_ALIASES.len() {
+            let contract_id_parameter_alias = &format!("--{}", &CONTRACT_ID_ALIASES[i]);
+            let add_key_args = AddKeyArgs::parse_from(&[
+                "near",
+                account_id,
+                access_key,
+                contract_id_parameter_alias,
+                contract_id,
+            ]);
+            let result = AddKeyArgs::to_cli_args(&add_key_args, "testnet".to_string());
+            assert_eq!(
+                result.join(" "),
+                format!("account add-key {} grant-function-call-access --allowance unlimited --contract-account-id {} --function-names  use-manually-provided-public-key {} network-config testnet sign-with-keychain send", account_id, contract_id, access_key)
+            )
+        }
+    }
+
+    #[test]
+    fn add_function_call_access_key_for_some_methods_testnet() {
+        let account_id = "bob.testnet";
+        let access_key = "ed25519:DReZmNmnGhpsYcCFFeYgPsJ9YCm9xH16GGujCPe3KQEq";
+        let contract_id = "example.testnet";
+        let method_names = "get,set";
+
+        for i in 0..METHOD_NAMES_ALIASES.len() {
+            let method_names_parameter_alias = &format!("--{}", &METHOD_NAMES_ALIASES[i]);
+            let contract_id_parameter_alias = &format!("--{}", &CONTRACT_ID_ALIASES[0]);
+            let add_key_args = AddKeyArgs::parse_from(&[
+                "near",
+                account_id,
+                access_key,
+                contract_id_parameter_alias,
+                contract_id,
+                method_names_parameter_alias,
+                method_names,
+            ]);
+            let result = AddKeyArgs::to_cli_args(&add_key_args, "testnet".to_string());
+            assert_eq!(
+                result.join(" "),
+                format!("account add-key {} grant-function-call-access --allowance unlimited --contract-account-id {} --function-names {} use-manually-provided-public-key {} network-config testnet sign-with-keychain send", account_id, contract_id, method_names, access_key)
```

Comment:
nit: It might be easier to read this way:

```suggestion
                format!("account add-key {account_id} grant-function-call-access --allowance unlimited --contract-account-id {contract_id} --function-names {method_names} use-manually-provided-public-key {access_key} network-config testnet sign-with-keychain send")
```